sihl_users/lib/service.ml

```diff
@@ -9,6 +9,9 @@ module User = struct
          "no user found, have you applied the authentication middleware?"
 
   let send_registration_email request user =
+    let (module Repository : Contract.REPOSITORY) =
+      Sihl_core.Registry.get Contract.repository
+    in
```

Comment:
After looking at this pattern, I'm not sure if I like this. Let's see if we can do something about it later.

Reply:
I don't like it either, adds a lot of noise. At some point we could write a ppx to have the right implementation injected.